test-data/unit/check-generics.test

```diff
@@ -1505,7 +1505,7 @@ T = TypeVar('T')
 class C(Generic[T]):
     def __init__(self) -> None: pass
 x = C # type: Callable[[], C[int]]
-y = C # type: Callable[[], int] # E: Incompatible types in assignment (expression has type C[T], variable has type Callable[[], int])
+y = C # type: Callable[[], int] # E: Incompatible types in assignment (expression has type Type[C[T]], variable has type Callable[[], int])
```

Comment:
Probably unrelated to this PR, but this error message looks not right, it should be ``Type[C]`` here (or maybe ``Type[C[Any]]``). If not and it is not easy to fix this (by just erasing the return type) in the same PR, could you please open a new issue?